hathor/verification/poa_block_verifier.py

```diff
@@ -36,22 +35,24 @@ def verify_poa(self, block: PoaBlock) -> None:
         poa_settings = self._settings.CONSENSUS_ALGORITHM
         assert isinstance(poa_settings, PoaSettings)
 
+        parent_block = block.get_block_parent()
+        if block.timestamp < parent_block.timestamp + self._settings.AVG_TIME_BETWEEN_BLOCKS:
+            raise PoaValidationError(
+                f'blocks must have at least {self._settings.AVG_TIME_BETWEEN_BLOCKS} seconds between them'
+            )
+
         # validate block rewards
         if block.outputs:
             raise PoaValidationError('blocks must not have rewards in a PoA network')
 
         # validate that the signature is valid
-        sorted_signers = sorted(poa_settings.signers)
-        signer_index: int | None = None
-
-        for i, public_key_bytes in enumerate(sorted_signers):
-            if self._verify_poa_signature(block, public_key_bytes):
-                signer_index = i
-                break
-
-        if signer_index is None:
+        signer = poa.get_signer_index_and_public_key(poa_settings, block.signer_id)
+        if signer is None:
             raise PoaValidationError('invalid PoA signature')
 
+        signer_index, public_key_bytes = signer
+        self._verify_poa_signature(block, public_key_bytes)
```

Comment:
You might want to assert that it returns True. It seems this call is not doing anything since it returns False if it fails.

Reply:
This has been changed. See https://github.com/HathorNetwork/hathor-core/pull/1071#discussion_r1669469069

Reply:
Ok. Thanks!